cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftStore.java

```diff
@@ -0,0 +1,301 @@
+                Fields.stringType(ID_COL),
+                Fields.stringType(ARTIFACT_COL),
+                Fields.stringType(NAME_COL),
+                Fields.stringType(DESCRIPTION_COL),
+                Fields.longType(CREATED_COL),
+                Fields.longType(UPDATED_COL),
+                Fields.stringType(PIPELINE_COL),
+                Fields.intType(REVISION_COL))
+    .withPrimaryKeys(NAMESPACE_COL, GENERATION_COL, OWNER_COL, ID_COL)
+    .build();
+  private static final Gson GSON = new Gson();
+  private final TransactionRunner transactionRunner;
+
+  public DraftStore(TransactionRunner transactionRunner) {
+    this.transactionRunner = transactionRunner;
+  }
+
+  /**
+   * @param namespace the namespace to fetch the drafts from
+   * @param owner the id of the owner of the drafts
+   * @param sortRequest The sorting that should be applied to the results, pass null if no sorting is required.
+   * @return a list of drafts
+   * @throws TableNotFoundException if the draft store table is not found
+   * @throws InvalidFieldException if the fields Namespace and owner fields do not match the fields in the
+   *   StructuredTable
+   */
+  public List<Draft> listDrafts(NamespaceSummary namespace, String owner,
+                                SortRequest sortRequest) throws TableNotFoundException, InvalidFieldException {
+    List<Field<?>> prefix = new ArrayList<>(3);
+    prefix.add(Fields.stringField(NAMESPACE_COL, namespace.getName()));
+    prefix.add(Fields.longField(GENERATION_COL, namespace.getGeneration()));
+    prefix.add(Fields.stringField(OWNER_COL, owner));
+
+    List<StructuredRow> rows;
+    rows = TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable table = context.getTable(TABLE_ID);
+      Range range = Range.singleton(prefix);
+      List<StructuredRow> temp = new ArrayList<>();
+      try (CloseableIterator<StructuredRow> rowIter = table.scan(range, Integer.MAX_VALUE)) {
+        while (rowIter.hasNext()) {
+          temp.add(rowIter.next());
+        }
+      }
+      return temp;
+    }, TableNotFoundException.class, InvalidFieldException.class);
+
+    List<StructuredRow> sortedResults = doSort(rows, sortRequest);
+    return sortedResults.stream().map(this::fromRow).collect(Collectors.toList());
+  }
+
+  /**
+   * Helper method to apply the sorting onto a list of rows. Sorting needs to take place at the store-level so it can
+   * leverage the StructuredRow to enable sorting on any field.
+   *
+   * @param rows list of {@link StructuredRow} to be sorted
+   * @param sortRequest {@link SortRequest} describing the sort to be performed
+   * @return a sorted list of {@link StructuredRow}
+   */
+  private List<StructuredRow> doSort(List<StructuredRow> rows, SortRequest sortRequest) {
+    if (sortRequest == null) {
+      return rows;
+    }
+    String sortField = sortRequest.getFieldName();
+    Map<String, FieldType.Type> fieldTypes = TABLE_SPEC
```

Comment:
Seems overkill to create a map to just get one field. It's better to do:

```
FieldType fieldType = TABLE_SPEC.getFieldTypes().stream().filter(f -> f.getName().equals(sortField)).findFirst().orElse(null);
if (fieldType == null) {
  throw ...
}
```

Reply:
done